tests/Feature/TagControllerTest.php

```diff
@@ -0,0 +1,171 @@
+use Tests\TestCase;
+
+
+class TagControllerTest extends TestCase
+{
+    use RefreshDatabase;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $user = User::factory()->create();
+        $adminRole = Role::where('name', 'admin')->first();
+
+        if (!$adminRole) {
+            // If the "admin" role does not exist, create it
+            $adminRole = Role::create(['name' => 'admin']);
+        }
+
+        $user->assignRole($adminRole);
+        $this->actingAs($user, 'api');
+    }
+
+    public function testIndexReturnsTags()
+    {
+        // Create some tags
+        TagFactory::new()->count(3)->create();
+
+        $response = $this->json('GET', '/api/v1/tags');
+
+        $response->assertStatus(200);
+
+        $response->assertJsonStructure([
+            'data' => [
+                '*' => [
+                    'id',
+                    'tag_name',
+                ],
+            ],
+        ]);
+    }
+
+    public function testIndexReturns404WhenNoTagsExist()
+    {
+        $response = $this->json('GET', '/api/v1/tags');
+
+        $response->assertStatus(404);
+
+        $response->assertJson(['message' => 'No hi ha tags a la base de dades']);
+    }
+
+    public function testStoreTag()
+    {
+        $tagData = Tag::factory()->make()->toArray();
+
+        $response = $this->json('POST', '/api/v1/tags', $tagData);
+
+        $response->assertStatus(201);
+
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'tag_name',
+            ],
+            'message',
+        ]);
+    }
+
+    public function testStoreFailsWhenTagNameIsMissing()
+    {
+        $response = $this->json('POST', '/api/v1/tags', []);
+
+        $response->assertStatus(422);
+
+        $response->assertJsonStructure([
+            'message',
+            'errors' => [
+                'tag_name'
+            ]
+        ]);
+
+        $response->assertJsonFragment([
+            'tag_name' => ['El camp tag name és obligatori.'],
+        ]);
+    }
+
+    public function testShowReturnsTag()
+    {
+        $tag = Tag::factory()->create();
+
+        $response = $this->json('GET', "/api/v1/tags/{$tag->id}");
+
+        $response->assertStatus(200);
+
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'tag_name',
+            ],
+        ]);
+
+        // Assert the response data matches the created tag
+        $response->assertJson([
+            'data' => [
+                'id' => $tag->id,
+                'tag_name' => $tag->tag_name,
+            ],
+        ]);
+    }
+
+    public function testUpdateTagSuccessfully()
+    {
+        $tag = Tag::factory()->create();
+
+        $updatedData = [
+            'tag_name' => 'Updated Tag Name',
+        ];
+
+        $response = $this->json('PUT', "/api/v1/tags/{$tag->id}", $updatedData);
+
+        $response->assertStatus(200);
```

Comment:
Shouldn't be another status code here? Same for L177.

Reply:
As we discussed, when not specifying the error on the tagController, still getting a 200. 201 is when u create a new register, maybe it's because the id already exists on the database, and it's not related to the tag_name directly.